RegionalTestApp/Program.cs

```diff
@@ -0,0 +1,107 @@
+ * 2. Verify the region was discovered "[Region discovery] Region: eastus"
+ * 3. Verify the token is obtained from the regional endpoint
+ * 4. You can use fiddler to verify the token endpoint
+ * 
+ * Case 2: Verify WithAzureRegion(false)
+ * 1. Verify the logs for "WithAzureRegion: False"
+ * 2. Verify the token is obtained from the global endpoint.
+ **/
+namespace TestApp
+{
+    class Program
+    {
+        static string clientId = "16dab2ba-145d-4b1b-8569-bf4b9aed4dc8";
+#pragma warning disable UseAsyncSuffix // Use Async suffix
+        static async Task Main(string[] args)
+#pragma warning restore UseAsyncSuffix // Use Async suffix
+        {
+            //Uncomment below line to test in dev env.
+            //Environment.SetEnvironmentVariable("REGION_NAME", "centralus");
+            X509Certificate2 certificate = ReadCertificate("97D8C9DB3C84874D0363DCA540778461B2291780");
+
+            Console.WriteLine("=== Acquire token regional ===");
+            await AcquireTokenAsync(true, certificate).ConfigureAwait(false);
+            Console.WriteLine();
+
+            Console.WriteLine("=== Acquire token global ===");
+            await AcquireTokenAsync(false, certificate).ConfigureAwait(false);
+
+        }
+
+        private static void Log(LogLevel level, string message, bool containsPii)
+        {
+            if (containsPii)
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+            }
+            Console.WriteLine($"{level} {message}");
+            Console.ResetColor();
+        }
+
+        private static X509Certificate2 ReadCertificate(string thumbprint)
+        {
+            if (string.IsNullOrWhiteSpace(thumbprint))
+            {
+                throw new ArgumentException("certificateName should not be empty. Please set the CertificateName setting in the appsettings.json", "certificateName");
+            }
+            X509Certificate2 cert = null;
+
+            using (X509Store store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            {
+                store.Open(OpenFlags.ReadOnly);
+                X509Certificate2Collection certCollection = store.Certificates;
+
+                // Find unexpired certificates.
+                X509Certificate2Collection currentCerts = certCollection.Find(X509FindType.FindByTimeValid, DateTime.Now, false);
+
+                // From the collection of unexpired certificates, find the ones with the correct name.
+                X509Certificate2Collection signingCert = currentCerts.Find(X509FindType.FindByThumbprint, thumbprint, false);
+
+                // Return the first certificate in the collection, has the right name and is current.
+                cert = signingCert.OfType<X509Certificate2>().OrderByDescending(c => c.NotBefore).FirstOrDefault();
+            }
+            return cert;
+        }
+
+        private static async Task AcquireTokenAsync(bool withAzureRegion, X509Certificate2 certificate)
```

Comment:
>bool withAzureRegion [](start = 52, length = 20)

move this parameter to the end and set it to false or true by default, so it's not needed, then you can remove either the true/false in line 43 or 47 and just pass the cert.